javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -57,6 +66,125 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
+            // grouping, possible with flattening first
+
+            // group example values into a map
+            Map<String, Object> exampleValue = new HashMap<>();
+            for (MethodTransformationDetail detail : convenienceMethod.getMethodTransformationDetails()) {
+                for (ParameterMapping parameterMapping : detail.getParameterMappings()) {
+                    if (parameterMapping.getOutputParameterPropertyName() != null) {
+                        // this is a flattened property, so put flattening(real parameter) value
+
+                        // output parameter's name is the "escaped reserved client method parameter name" of the real parameter's serialized name
+                        // since flattened parameter is always in body, we can deal with that explicitly
+                        ClientMethodParameter outputParameter = detail.getOutParameter();
+                        Map<String, Object> flattenedParameterValue = findFlattenedBodyParameterValue(proxyMethodExample, outputParameter.getName());
+                        if (flattenedParameterValue != null) {
+                            exampleValue.putAll(flattenedParameterValue);
+                        }
+                        // since it's flattened property, all parameterMappings share the same outputParameter(real parameter)
+                        // we only need to put example value once, which is the real parameter's value
+                        break;
+                    } else {
+                        // Group property's "serializedName" is the real parameter's "serializedName" on the wire.
+                        // This implicit equivalence is defined in emitter and preserved in mapping client method.
+                        String serializedParameterName = parameterMapping.getInputParameterProperty().getSerializedName();
+                        ClientMethodParameter parameter = detail.getOutParameter();
+                        exampleValue.put(serializedParameterName,
+                                ModelExampleUtil.getParameterExampleValue(
+                                        proxyMethodExample, serializedParameterName, parameter.getRequestParameterLocation()));
+                    }
+                }
+            }
+            IType type = methodParameter.getClientMethodParameter().getClientType();
+            IType wireType = methodParameter.getClientMethodParameter().getWireType();
+            return ModelExampleUtil.parseNode(type, wireType, exampleValue);
+        } else if (isFlattenParameter(convenienceMethod, methodParameter)) {
+            // flatten, no grouping
+            String realParameterName = convenienceMethod.getMethodTransformationDetails().iterator().next().getOutParameter().getName();
+            Map<String, Object> realParameterValue = (Map<String, Object>) proxyMethodExample.getParameters().get(realParameterName).getObjectValue();
```

Comment:
should it also call `findFlattenedBodyParameterValue`?

Reply:
Yes it should. Thanks!

Reply:
Changed.